SU2_CFD/include/numerics/turbulent/transition/trans_diffusion.hpp

```diff
@@ -103,3 +103,78 @@ class CAvgGrad_TransLM final : public CAvgGrad_Scalar<FlowIndices> {
   }
 
 };
+
+
+/*!
+ * \class CAvgGrad_TransAFT
+ * \brief Class for computing viscous term using average of gradient with correction (AFT transition model).
+ * \ingroup ViscDiscr
+ * \author S. Kang.
+ */
+template <class FlowIndices>
+class CAvgGrad_TransAFT final : public CAvgGrad_Scalar<FlowIndices> {
+private:
+  using Base = CAvgGrad_Scalar<FlowIndices>;
+  using Base::Laminar_Viscosity_i;
+  using Base::Laminar_Viscosity_j;
+  using Base::Eddy_Viscosity_i;
+  using Base::Eddy_Viscosity_j;
+  using Base::Density_i;
+  using Base::Density_j;
+  using Base::ScalarVar_i;
+  using Base::ScalarVar_j;
+  using Base::Proj_Mean_GradScalarVar;
+  using Base::proj_vector_ij;
+  using Base::Flux;
+  using Base::Jacobian_i;
+  using Base::Jacobian_j;
+
+  /*!
+   * \brief Adds any extra variables to AD
+   */
+  void ExtraADPreaccIn() override {}
+
+  /*!
+   * \brief LM transition model specific steps in the ComputeResidual method
+   * \param[in] config - Definition of the particular problem.
+   */
+  void FinishResidualCalc(const CConfig* config) override {
+    const bool implicit = config->GetKind_TimeIntScheme() == EULER_IMPLICIT;
+
+    /*--- Compute mean effective dynamic viscosity ---*/
+    const su2double diff_i_AF = Laminar_Viscosity_i + Eddy_Viscosity_i;
+    const su2double diff_j_AF = Laminar_Viscosity_j + Eddy_Viscosity_j;
+    const su2double diff_i_Gamma = Laminar_Viscosity_i + Eddy_Viscosity_i;
+    const su2double diff_j_Gamma = Laminar_Viscosity_j + Eddy_Viscosity_j;
+
+    const su2double diff_AF = 0.5*(diff_i_AF + diff_j_AF);
+    const su2double diff_Gamma = 0.5*(diff_i_Gamma + diff_j_Gamma);
+
+    Flux[0] = diff_AF*Proj_Mean_GradScalarVar[0];
+    Flux[1] = diff_Gamma*Proj_Mean_GradScalarVar[1];
+
+    /*--- For Jacobians -> Use of TSL (Thin Shear Layer) approx. to compute derivatives of the gradients ---*/
+    if (implicit) {
+      const su2double proj_on_rho_i = proj_vector_ij/Density_i;
+      Jacobian_i[0][0] = -diff_AF*proj_on_rho_i;  Jacobian_i[0][1] = 0.0;
+      Jacobian_i[1][0] = 0.0;                        Jacobian_i[1][1] = -diff_Gamma*proj_on_rho_i;
+
+      const su2double proj_on_rho_j = proj_vector_ij/Density_j;
+      Jacobian_j[0][0] = diff_AF*proj_on_rho_j;   Jacobian_j[0][1] = 0.0;
+      Jacobian_j[1][0] = 0.0;                     Jacobian_j[1][1] = diff_Gamma*proj_on_rho_j;
+    }
+  }
```

Comment:
This only differs from LM by some constant factors, refactor it to use the same class.